token-swap/src/index.ts

```diff
@@ -1,2 +1,120 @@
-import * as ts from '@solana/spl-token-swap';
-import * as w3 from '@solana/web3.js'
+import { Numberu64, TokenSwap } from "@solana/spl-token-swap";
+import { Connection, PublicKey, Account } from "@solana/web3.js";
+
+/**
+ * An Orca token-swap pool
+ * @param address The pool account address
+ * @param nonce The nonce used to generate the pool authority
+ * @param authority The pool authority PDA address to sign instructions
+ * @param poolTokenAddress Public address for the pool token
+ * @param feeAccount Public address of the pool token fee account
+ * @param tokenNameA Name of Token A
+ * @param tokenAccountA Public address of Token A within this pool
+ * @param tokenNameB Name of Token B
+ * @param tokenAccountB Public address of Token B within this pool
+ * @param curveType Trading curve type. 0 - ConstantProduct, 1 - ConstantPrice, 3 - Offset
+ */
+export type OrcaPool = {
+    address: PublicKey
+    nonce: number
+    authority: PublicKey
+    poolTokenAddress: PublicKey
+    feeAccount: PublicKey
+    tokenNameA: String
+    tokenAccountA: PublicKey
+    tokenNameB: String
+    tokenAccountB: PublicKey
+    curveType: number
+}
+
+/**
+ * Instructions for performing a swap
+ *
+ * @param userSource User's source token A account
+ * @param userDestination User's destination token B account
+ * @param amountIn Amount of token A to transfer from source account
+ * @param minimumAmountOut Minimum amount of token B the user will receive
+ */
+export type OrcaSwapInstruction = {
+    pool: OrcaPool,
+    sourceTokenAddress: PublicKey,
+    destinationTokenAddress: PublicKey,
+    amountIn: number | Numberu64,
+    minimumAmountOut: number | Numberu64,
+}
+
+export class Orca {
+
+    /**
+     * Swap token A for token B
+     * @param connection Web3 Solana Connection Object
+     * @param userTransferAuthority Account delegated to transfer user's tokens
+     * @param feePayer Account designated to pay the fee of the swap
+     * @param instruction Instructions for the swap
+     * @throws Error types:
+     *      - Unable to load the orca token swap pool
+     *      - Authority error in the swap instruction
+     *      - Token A & B does not match the input orca pool
+     *      - Insufficent balance in user source account or fee account
+     *      - Invalid public address input
+     */
+    static async swap(
+        connection: Connection,
+        swapInstruction: OrcaSwapInstruction,
+        userTransferAuthority: Account,
+        feePayer: Account) {
+
+        /** Untested Sample Code!! */
+        let poolInfo = swapInstruction.pool
+        let poolTokenSwapAccount: PublicKey = poolInfo.address
+        let tokenSwapProgramId: PublicKey = new PublicKey(0)
+        let tokenSwap = await TokenSwap.loadTokenSwap(connection, poolTokenSwapAccount, tokenSwapProgramId, feePayer).catch(e => {
```

Comment:
Is the code block inside "Untested Sample Code" supposed to be the implementation? Also, is the idea to create a user-facing interface that's simpler than what already exists in `token-swap`? If so, I'm all for it. My suggestion then would be to consider ditching `loadTokenSwap` and `swap` functions altogether.

Reply:
Just to make sure we're aligned on the goal of the sdk project - my intention is to make this easier for the developers of the web/nodejs users that are interacting with Orca as well. As a beginner to the blockchain space, I actually found it quite difficult to understand the token-swap sdk on the first go, so I think we can help by building a simpler API & more beginner-friendly documentation. 

This is where I might need your input here. I'm envisioning this as a wrapper above token-swap with some helper types that are more descriptive than the token-swap counterparts. Is there a reason why we would not depend on token-swap's functions?